FormalConjectures/ErdosProblems/107.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 107
+
+*Reference:* [erdosproblems.com/107](https://www.erdosproblems.com/107)
+-/
+
+open Filter
+
+namespace Erdos107
+
+abbrev Point := EuclideanSpace ℝ (Fin 2)
+
+namespace Point
+
+/-- The three points `a,b,c` are not collinear. -/
+def InGenPos₃ (a b c : Point) : Prop :=
+  Matrix.det !![a 0, b 0, c 0 ; a 1, b 1, c 1 ; 1, 1, 1] ≠ 0
+
+/-- No three points in `S` are collinear. -/
+def SetInGenPos (S : Set Point) : Prop :=
+  ∀ ⦃p q r : Point⦄, {p,q,r} ⊆ S → List.Nodup [p,q,r] → InGenPos₃ p q r
+
+end Point
+
+/-- The set `S` is in convex position. -/
+def ConvexIndep (S : Set Point) : Prop :=
+  ∀ a ∈ S, a ∉ convexHull ℝ (S \ {a})
+
+/-- The set `P` contains a convex `n`-gon. -/
+def HasConvexNGon (n : ℕ) (P : Set Point) : Prop :=
```

Comment:
perhaps **https://github.com/google-deepmind/formal-conjectures/blob/603575bec6867fa91ab460c7065e83aa00906e7f/FormalConjectures/ForMathlib/Geometry/2d.lean#L51** could be used here?

Reply:
It could (or maybe the `IsConvexPolygon` variant), but I slightly prefer the current formulation because (1) `Fin n -> P` has the extra structure of an ordering on `p`, yet abstractly we just want to talk about sets of points and (2) going between "all angles are CCW" and "the set is convex" requires some proof.

Would you prefer `Is(Ccw)ConvexPolygon` despite this?

Reply:
I think the current formulation is reasonable! (in fact I think it would be reasonable to add it to `ForMathlib` since it's useful when we don't want to assume that extra structure)